airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/interpolated_nested_mapping.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from dataclasses import InitVar, dataclass
+from typing import Any, Mapping, Optional, Union
+
+from airbyte_cdk.sources.declarative.interpolation.jinja import JinjaInterpolation
+from airbyte_cdk.sources.declarative.types import Config
+
+NestedMappingEntry = Union[dict[str, "NestedMapping"], list["NestedMapping"], str, int, float, bool, None]
+NestedMapping = Union[dict[str, NestedMappingEntry], str]
+
+
+@dataclass
+class InterpolatedNestedMapping:
+    """
+    Wrapper around a nested dict which can contain lists and primitive values where both the keys and values are interpolated recursively.
+
+    Attributes:
+        mapping (NestedMapping): to be evaluated
+    """
+
+    mapping: NestedMapping
+    parameters: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, parameters: Optional[Mapping[str, Any]]):
+        self._interpolation = JinjaInterpolation()
+        self._parameters = parameters
+
+    def eval(self, config: Config, **additional_parameters):
+        return self._eval(self.mapping, config, **additional_parameters)
+
+    def _eval(self, value, config, **kwargs):
+        # Recursively interpolate dictionaries and lists
+        if type(value) == str:
```

Comment:
nit: `isinstance` is generally a better pattern to use that comparing the type with equality to support inheritance https://stackoverflow.com/a/1549854